### PR TITLE
Fix EDoc

### PR DESCRIPTION
**Pull request by erszcz**
This PR addresses the doc comments which are unparseable by EDoc.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[7453.1](https://travis-ci.org/esl/MongooseIM/jobs/613576549) / Erlang 22.0 / small_tests / a397ff5a13121af25363ec75cb02a22efa3b3869
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2537/7453/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2537/7453/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-11-18_16.20.13/lib.mongooseim.logs/run.2019-11-18_16.20.13/suite.log.html)

File: src/amp.erl [hunk 1]
```diff
@@ -1,8 +1,5 @@
 -module(amp).
 %% @doc Models and business logic for  XEP-0079: Advanced Message Processing
-%% @reference <a href="http://xmpp.org/extensions/xep-0079.html">XEP-0079</a>
-%% @author <mongooseim@erlang-solutions.com>
-%% @copyright 2014 Erlang Solutions, Ltd.
```
Comment (erszcz) [inline src/amp.erl]: This needs a rewrite, not just removal.

File: src/amp_strategy.erl [hunk 2]
```diff
@@ -1,9 +1,6 @@
 -module(amp_strategy).
 %% @doc This module is responsible for determining the server's strategy for a
 %% particular message. (See XEP section 2.2.2 "Determine Default Action")
-%% @reference <a href="http://xmpp.org/extensions/xep-0079.html">XEP-0079</a>
-%% @author <mongooseim@erlang-solutions.com>
-%% @copyright 2014 Erlang Solutions, Ltd.
```
Comment (erszcz) [inline src/amp_strategy.erl]: To rewrite.

File: src/mod_amp.erl [hunk 3]
```diff
@@ -1,8 +1,5 @@
 -module(mod_amp).
 %% @doc MongooseIM/Ejabberd module for (a subset of) XEP-0079 support.
-%% @reference <a href="http://xmpp.org/extensions/xep-0079.html">XEP-0079</a>
-%% @author <mongooseim@erlang-solutions.com>
-%% @copyright 2014 Erlang Solutions, Ltd.
```
Comment (erszcz) [inline src/mod_amp.erl]: To rewrite.

**Comment by mongoose-im** [comment]
[7454.1](https://travis-ci.org/esl/MongooseIM/jobs/613576984) / Erlang 22.0 / small_tests / e80dfff2c44d5c62404a3f3eac30c96c51e33666
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2537/7454/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2537/7454/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-11-18_17.25.50/lib.mongooseim.logs/run.2019-11-18_17.25.50/suite.log.html)

**Comment by erszcz** [comment]
@michalwski I was using an [experimental EDoc plugin](https://github.com/erszcz/edoc/tree/import-docs-chunks) to make chunks for ExDoc, but the same can be checked just by running `rebar3 edoc`.

**Comment by erszcz** [comment]
Hmmm, `rebar3 edoc` seems to be even more strict - it still doesn't pass even with these fixes applied :/

**Comment by mongoose-im** [comment]
[7456.1](https://travis-ci.org/esl/MongooseIM/jobs/613902733) / Erlang 22.0 / small_tests / c26f0ac9a26f6c553b1226ec054fa0a31f806d63
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2537/7456/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2537/7456/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-11-19_09.04.29/lib.mongooseim.logs/run.2019-11-19_09.04.29/suite.log.html)

**Comment by michalwski** [comment]
Thanks for the fixes @erszcz I'm going to merge to PR as when CI passes (although this is only documentation change). 

I wonder if it would make sense to add `rebar3 edoc` to our scripts on CI and fail the build if there is sth wrong with the inline documentation. What do you think?

**Comment by erszcz** [comment]
> I wonder if it would make sense to add rebar3 edoc to our scripts on CI and fail the build if there is sth wrong with the inline documentation. What do you think?

Maybe not fail the build, but at least have a clear indication that the docs aren't OK anymore.

**Comment by fenek** [comment]
> > I wonder if it would make sense to add rebar3 edoc to our scripts on CI and fail the build if there is sth wrong with the inline documentation. What do you think?
> 
> Maybe not fail the build, but at least have a clear indication that the docs aren't OK anymore.

I actually think that it is OK to make it a part of `dialyzer_only` job.

**Comment by michalwski** [comment]
> clear indication that the docs aren't OK

From my experience the clearest indication that sth is not OK is when a build is marked with red color and tagged as `failed`. Otherwise sooner or later things are starting being ignored if eve they are yellow and nod green :)

**Comment by michalwski** [comment]
Making it as part of dialyzer job in a separate PR is definitely a good idea.

**Pull request merged**